python/lsst/validate/drp/matchreduce.py

```diff
@@ -268,7 +313,7 @@ def _loadAndMatchCatalogs(repo, dataIds, matchRadius,
         tmpCat['base_PsfFlux_snr'][:] = tmpCat['base_PsfFlux_instFlux'] \
             / tmpCat['base_PsfFlux_instFluxErr']
 
-        if useJointCal:
+        if doApplyExternalWcs:
             for record in tmpCat:
                 record.updateCoord(wcs)
```

Comment:
I know you didn't touch this line, but please replace this loop with `afw.table.wcsUtils.updateSourceCoords(wcs, tmpCat)`, which does the loop in C++.